### PR TITLE
Bugfix MTE-2679 Firefox iOS full functional test plan for iOS 15

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-2679)

## :bulb: Description
I have disabled the tests that don't work with iOS 15, added iOS 15 specific commands and adjusted commands that work on multiple iOS versions.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BrowsingPDFTests.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ class BrowsingPDFTests: BaseTestCase {
         app.swipeUp()
         mozWaitForElementToExist(app.staticTexts["1 of 1"])
         app.swipeDown()
-        XCTAssertTrue(app.staticTexts["1 of 1"].exists)
+        mozWaitForElementToExist(app.staticTexts["1 of 1"])
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/BrowsingPDFTests.swift:25]: `XCTAssertTrue(...exists)` may not work on iOS 15 all the time.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/CreditCardsTests.swift [hunk 2]
```diff
@@ -26,10 +26,7 @@ class CreditCardsTests: BaseTestCase {
 
     // https://testrail.stage.mozaws.net/index.php?/cases/view/2306967
     // SmokeTest
-    func testAccessingTheCreditCardsSection() throws {
-        if #unavailable(iOS 16) {
-            throw XCTSkip("Test fails intermittently for iOS 15")
-        }
+    func testAccessingTheCreditCardsSection() {
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/CreditCardsTests.swift:29]: I tried again and some smoke tests should work on iOS 15 as well. 😃

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/CreditCardsTests.swift [hunk 3]
```diff
@@ -125,11 +125,7 @@ class CreditCardsTests: BaseTestCase {
 
     // https://testrail.stage.mozaws.net/index.php?/cases/view/2306969
     // Smoketest
-    func testAutofillCreditCardsToggleOnOoff() throws {
-        if #unavailable(iOS 16) {
-            throw XCTSkip("Test fails intermittently for iOS 15")
-        }
-        // Disable the "Save and Fill Payment Methods" toggle
+    func testAutofillCreditCardsToggleOnOoff() {
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/CreditCardsTests.swift:128]: Same here. This smoke test should work on iOS 15.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/CreditCardsTests.swift [hunk 4]
```diff
@@ -142,40 +138,45 @@ class CreditCardsTests: BaseTestCase {
...
-        waitForExistence(app.buttons["Done"])
-        app.buttons["Done"].tap()
-        app.swipeUp()
-        mozWaitForElementToExist(app.webViews["contentView"].webViews.staticTexts["Explore Checkout"])
-        mozWaitForElementToExist(cardNumber)
-        cardNumber.tapOnApp()
-        // The autofill option (Use saved card prompt) is displayed
-        if !app.buttons[useSavedCard].exists {
-            app.webViews["contentView"].webViews.textFields["Full name on card"].tapOnApp()
+        if #available(iOS 16, *) {
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/CreditCardsTests.swift:141]: On iOS 15, the test has trouble going back to the new tab screen.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/DownloadsTests.swift [hunk 5]
```diff
@@ -118,9 +118,13 @@ class DownloadsTests: BaseTestCase {
...
         mozWaitForElementToExist(app.tables["DownloadsTable"].staticTexts[testFileNameDownloadPanel])
-        mozWaitForElementToExist(app.collectionViews.cells["Copy"])
+        if #available(iOS 16, *) {
+            mozWaitForElementToExist(app.collectionViews.cells["Copy"])
+        } else {
+            mozWaitForElementToExist(app.collectionViews.buttons["Copy"])
+        }
         if !iPad() {
-            app.otherElements.element(boundBy: 1).buttons["Close"].tap()
+            app.navigationBars["UIActivityContentView"].buttons["Close"].tap()
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/DownloadsTests.swift:127]: `app.navigationBars["UIActivityContentView"].buttons["Close"]` works on iOS 15, 16 and 17.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/ThirdPartySearchTest.swift [hunk 6]
```diff
@@ -104,11 +109,16 @@ class ThirdPartySearchTest: BaseTestCase {
...
+        if #unavailable(iOS 16) {
+            // Wait for "Fennec pasted from XCUITests-Runner" banner to disappear
+            sleep(2)
+        }
     }
 
     private func dismissSearchScreen() {
         mozWaitForElementToExist(app.navigationBars["Search"].buttons["Settings"])
         app.navigationBars["Search"].buttons["Settings"].tap()
+        mozWaitForElementToExist(app.navigationBars["Settings"].buttons[AccessibilityIdentifiers.Settings.navigationBarItem])
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/ThirdPartySearchTest.swift:121]: `mozWaitForElementToExist` works on iOS 15, 16 and 17.

**Review by isabelrios — APPROVED** [review]

**Pull request merged**